recipes/userver/all/conanfile.py

```diff
@@ -0,0 +1,655 @@
+import os
+
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import get, copy, export_conandata_patches, apply_conandata_patches
+from conan.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.53.0"
+
+class UserverConan(ConanFile):
+    name = 'userver'
+    description = 'The C++ Asynchronous Framework'
+    topics = ('framework', 'coroutines', 'asynchronous')
+    url = 'https://github.com/conan-io/conan-center-index'
+    homepage = 'https://userver.tech/'
+    license = 'Apache-2.0'
+    package_type = "library"
+
+    settings = 'os', 'arch', 'compiler', 'build_type'
+    options = {
+        'shared': [True, False],
+        'fPIC': [True, False],
+        'lto': [True, False],
+        'with_jemalloc': [True, False],
+        'with_mongodb': [True, False],
+        'with_postgresql': [True, False],
+        'with_postgresql_extra': [True, False],
+        'with_redis': [True, False],
+        'with_grpc': [True, False],
+        'with_clickhouse': [True, False],
+        'with_rabbitmq': [True, False],
+        'with_utest': [True, False],
+        'namespace': ['ANY'],
+        'namespace_begin': ['ANY'],
+        'namespace_end': ['ANY'],
+    }
+
+    default_options = {
+        'shared': False,
+        'fPIC': True,
+        'lto': True,
+        'with_jemalloc': False,
+        'with_mongodb': True,
+        'with_postgresql': True,
+        'with_postgresql_extra': False,
+        'with_redis': True,
+        'with_grpc': True,
+        'with_clickhouse': True,
+        'with_rabbitmq': True,
+        'with_utest': True,
+        'namespace': 'userver',
+        'namespace_begin': 'namespace userver {',
+        'namespace_end': '}',
+        'mongo-c-driver/*:with_sasl': 'cyrus',
+    }
+
+    @property
+    def _source_subfolder(self):
+        return 'source'
+
+    @property
+    def _build_subfolder(self):
+        return os.path.join(self.build_folder, 'userver')
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires('boost/1.79.0', transitive_headers=True)
+        self.requires('c-ares/1.19.1')
+        self.requires('cctz/2.3', transitive_headers=True)
+        self.requires('concurrentqueue/1.0.3', transitive_headers=True)
+        self.requires('cryptopp/8.7.0')
+        self.requires('fmt/8.1.1', transitive_headers=True)
+        self.requires('libnghttp2/1.51.0')
+        self.requires('libcurl/7.88.1')
+        self.requires('libev/4.33')
+        self.requires('http_parser/2.9.4')
+        self.requires("openssl/[>=1.1 <4]")
+        self.requires('rapidjson/cci.20220822', transitive_headers=True)
+        self.requires('yaml-cpp/0.7.0')
+        self.requires("zlib/[>=1.2.11 <2]")
+
+        if self.options.with_jemalloc:
```

Comment:
While we do some amount of local testing for options like these to ensure that they compile, we might miss some (Specially combinations of options), so it would be really helpful to get your local compilation logs from adding these, it can again aid in review and future maintenance work _a lot_! Thanks :)